src/witan/app/handler.clj

```diff
@@ -44,7 +44,10 @@
 (defn add-token!
   [user-id tokens]
   (let [token (user/random-token)
-        ttl (t/days 28)]
+        ttl (t/days 28)
+        existing (some #(if (= user-id (-> % second :user)) %) @tokens)]
+    (when existing ;; remove an existing token for this user
+      (swap! tokens dissoc (first existing)))
```

Comment:
yeah, this will become a pain quickly. There is an example of using a C* session store here:

https://github.com/MastodonC/kixi.hecuba/blob/master/src/clj/kixi/hecuba/session.clj